fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -13,280 +8,321 @@
+
+                    ArrayAdapter<String> arrayAdapter =
+                            new ArrayAdapter<>(context, android.R.layout.select_dialog_singlechoice);
+
+                    arrayAdapter.add("Roll-Call Event");
+                    arrayAdapter.add("Election Event");
+
+                    builder.setNegativeButton("Cancel", (dialog, which) -> dialog.dismiss());
+                    builder.setAdapter(
+                            arrayAdapter,
+                            ((dialog, which) -> {
+                                dialog.dismiss();
+                                viewModel.chooseEventType(EventType.values()[which]);
+                            }));
+                    builder.show();
+                };
+
+        binding.setIsFutureCategory(eventCategory.equals(FUTURE));
+        binding.setViewmodel(viewModel);
+        binding.setLifecycleOwner(lifecycleOwner);
+        binding.setAddEventListener(addEventOnClickListener);
+        binding.executePendingBindings();
+
+        binding.addFutureEventButton.setFocusable(false);
+
+        return binding.getRoot();
+    }
+
+    /**
+     * Gets a View that displays the data for the given child within the given group.
+     *
+     * @param groupPosition the position of the group that contains the child
+     * @param childPosition the position of the child (for which the View is returned) within the
+     *     group
+     * @param isLastChild Whether the child is the last child within the group
+     * @param convertView the old view to reuse, if possible. You should check that this view is
+     *     non-null and of an appropriate type before using. If it is not possible to convert this
+     *     view to display the correct data, this method can create a new view. It is not guaranteed
+     *     that the convertView will have been previously created by {@link #getChildView(int, int,
+     *     boolean, View, ViewGroup)}.
+     * @param parent the parent that this view will eventually be attached to
+     * @return the View corresponding to the child at the specified position
+     */
+    @Override
+    public View getChildView(
+            int groupPosition,
+            int childPosition,
+            boolean isLastChild,
+            View convertView,
+            ViewGroup parent) {
+
+
+        ViewDataBinding binding;
+        Event event = ((Event) getChild(groupPosition, childPosition));
+        if (convertView == null) {
+            LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+            binding = LayoutEventBinding.inflate(inflater, parent, false);
+        } else {
+            binding = DataBindingUtil.getBinding(convertView);
+        }
+
+        EventCategory category = (EventCategory) getGroup(groupPosition);
+
+        // we use a switch case to handle all the different type of actions we want when we click on a certain event
+
+        switch (event.getType()) {
+            case ELECTION:
```

Comment:
You can refactor everything inside this case into a method, this way the method `getChildView` doesn't become extremely long with other events arrive